src/Sylius/Bundle/CoreBundle/PaymentRequest/CommandDispatcher/PaymentRequestCommandDispatcher.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\PaymentRequest\CommandDispatcher;
+
+use Sylius\Bundle\CoreBundle\PaymentRequest\CommandProvider\PaymentRequestCommandProviderInterface;
+use Sylius\Bundle\PaymentBundle\Exception\PaymentRequestNotSupportedException;
+use Sylius\Component\Payment\Model\PaymentRequestInterface;
+use Sylius\Component\Payment\Repository\PaymentRequestRepositoryInterface;
+use Symfony\Component\Messenger\MessageBusInterface;
+
+final class PaymentRequestCommandDispatcher implements PaymentRequestCommandDispatcherInterface
+{
+    public function __construct(
+        private PaymentRequestRepositoryInterface $paymentRequestRepository,
+        private PaymentRequestCommandProviderInterface $paymentRequestCommandProvider,
+        private MessageBusInterface $commandBus,
+    ) {
+    }
+
+    public function add(PaymentRequestInterface $paymentRequest): void
+    {
+        $this->paymentRequestRepository->add($paymentRequest);
```

Comment:
this logic will persist new object and commit transaction. Do we really want to do it at this stage? Shouldn't we rather just persist it and depend on flushing around command bus?

All in all, maybe this logic should be based more on commands?

Reply:
To some extend logic in this class could be simplify to one function, where we will always `persist()` on object manager

Reply:
Reading the code of promotion catalog, I see how to solve this.
Lets just makee the same as here : https://github.com/Sylius/Sylius/blob/1.13/src/Sylius/Bundle/ApiBundle/CommandHandler/Catalog/AddProductReviewHandler.php#L68
And then I will add an event subscriber like this one dispatching the commands :
https://github.com/Sylius/Sylius/blob/1.13/src/Sylius/Bundle/ApiBundle/EventSubscriber/CatalogPromotionEventSubscriber.php#L47